plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -863,6 +863,54 @@ public List<String> getAllViews(String databaseName)
         }
     }
 
+    @Override
+    public Optional<List<SchemaTableName>> getAllTables()
+    {
+        try {
+            return retry()
+                    .stopOn(UnknownDBException.class)
+                    .stopOnIllegalExceptions()
+                    .run("getAllTables", stats.getGetAllTables().wrap(() -> {
+                        try (ThriftMetastoreClient client = createMetastoreClient()) {
+                            return client.getAllTables();
+                        }
+                    }));
+        }
+        catch (TException e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, e);
+        }
+        catch (Exception e) {
+            throw propagate(e);
+        }
+    }
+
+    @Override
+    public Optional<List<SchemaTableName>> getAllViews()
+    {
+        // Without translateHiveViews, Hive views are represented as tables in Trino,
+        // and they should not be returned from ThriftHiveMetastore.getAllViews() call
+        if (!translateHiveViews) {
+            return Optional.empty();
+        }
```

Comment:
the impl of this method differs from `getAllViews(String databaseName)`. that one seems to return tables with `PRESTO_VIEW_FLAG` set to true when translation is disabled instead of empty result. 

was this intentional? Or am I misreading the code?

Reply:
I think the initial implementation was supposed to work only when translateHiveViews or when !translateHiveViews (not sure which one).